### ISSUE TITLE
optional PUT, DELETE request (for upload files)

### ISSUE THREAD

**Issue by TomS-** [kzahel/web-server-chrome#135]
I can PUT files and that works but DELETE does nothing. I've tried doing a catch for an error return and nothing is returned.

I've checked handler.js and nothing exists for removing files only replacing. 

Am I missing something or is "DELETE" not an option?

**Comment by kzahel** [comment]
Right. I don't think I ever implemented DELETE. I guess that would be a separate advanced option, "allow deleting files" (instead of just allow file upload)

**Comment by TomS-** [comment]
@kzahel is there anyway I can do this in the JS? I'm building a web-app that is being used as a demo tool. It's going out on next Friday and the client has requested a delete option.

**Comment by kzahel** [comment]
It shouldn't be too hard to add in support. I would be open to doing it but don't have much time in the next week. Maybe you could give it a shot to implement just the back-end yourself?

**Comment by er2er** [comment]
For "DELETE" support (in most simple form) you need to add the following method for DirectoryEntryHandler in handlers.js:
```
        delete: function() {
            this.fs.getByPath(this.request.path, (entry) => {
                entry.remove(()=>{
                    this.responseLength = 0
                    this.writeHeaders(200)
                    this.finish()
                });
            });
        },
```

Then, enable "DELETE" method in connection.js by adding it to the following check:
```
            if (['GET','HEAD','PUT','OPTIONS','DELETE'].includes(method)) {
                this.onRequest(this.curRequest)
            } else {
                console.error('don\'t know how to handle',this.curRequest)
            }

```

**Comment by kzahel** [comment]
Already implemented in the new version — the CLI supports file uploads via PUT/POST with the `--upload` flag. Sign up at https://ok200.app for release updates.
